### PR TITLE
[pysa][torch] Skip the assertion on the return type when the annotation is a forward reference

### PR DESCRIPTION
**Pull request by ZainRizvi**
Summary:
The assertion is causing build failures when running Pysa, our security-focused static analyzer.
This is because we run `pyre infer` on the source code before analyzing it, which introduces annotations such as `def foo() -> 'torch._tensor.Tensor'`.
This does not work with the `out_wrapper` decorator which relies on inspecting the signature of the decorated function.
Let's skip the check on the return type if we detect that it was introduced by `pyre infer`.

Test Plan:
```
buck2 'bxl' '--client-metadata' 'id=pyre' 'prelude//python/sourcedb/classic.bxl:build' '--console=simple' '--' '--target' 'fbcode//frl/ctrl/src2-main/platform/pybmi/pybmi/modeling/label_inference/scripts:query_rp_for_cc_datasets' '--target' 'fbcode//frl/ctrl/src2-main/platform/pybmi/doc:sphinx' '--target' 'fbcode//frl/ctrl/src2-main/platform/pybmi/pybmi/modeling/emg_pose:bento_kernel_ctrl_emg_pose' '--target' 'fbcode//frl/ctrl/src2-main/platform/pybmi/doc:make_tutorials' '--target' 'fbcode//frl/ctrl/src2-main/platform/workflow/workflow/components/cc_fa:failure_analysis_example_factory' '--target' 'fbcode//frl/ctrl/pte/environment:pte_all_deps_base'
```
on top of D50807445 succeeds.

Reviewed By: tianhan0

Differential Revision: D50838601

### CODE REVIEW

File: torch/_prims_common/wrappers.py [hunk 1]
```diff
@@ -298,7 +298,19 @@ def _fn(*args, out=None, **kwargs):
...
+        ):
+            # FIXME(T168485992): This is a forward reference, i.e a type
+            # annotation wrapped in a string literal. In theory, we would need
+            # to parse it, but we don't have the global context (i.e imports)
+            # to do that correctly. Since we know it is most likely introduced
+            # by `pyre infer` before running Pysa, so we can safely ignore it
+            # for now.
+            pass
+        else:
+            assert sig.return_annotation in (sig.empty, out_type)
```
Comment (lezcano) [inline torch/_prims_common/wrappers.py:313]: Annotations can be strings in some cases, which is fair, so what about simply
```suggestion
        assert isinstance(sig.return_annotation, str) or sig.return_annotation in (sig.empty, out_type)
```
This is a bit more forward-looking as it also covers functions thay may not return tensors.

Comment (ZainRizvi) [reply torch/_prims_common/wrappers.py:313]: For reference, this diff was unfortunately landed internally a couple days ago and there's a followup diff which adds annotations D50847843

Comment (lezcano) [reply torch/_prims_common/wrappers.py:313]: I'm not completely sure what these things imply

**Pull request closed without merging**